src/DataMapper/Observer/RelationObserver.php

```diff
@@ -0,0 +1,205 @@
+<?php
+/**
+ * Part of Windwalker project. 
+ *
+ * @copyright  Copyright (C) 2015 {ORGANIZATION}. All rights reserved.
+ * @license    GNU General Public License version 2 or later;
+ */
+
+namespace Windwalker\DataMapper\Observer;
+
+use JObservableInterface;
+use JObserverInterface;
+use Windwalker\Data\DataSet;
+use Windwalker\DI\Container;
+use Windwalker\Relation\Handler\AbstractRelationHandler;
+use Windwalker\Relation\Relation;
+use Windwalker\Table\Table;
+use Windwalker\Utilities\ArrayHelper;
+
+/**
+ * An observer to help DataMapper handle relations.
+ * 
+ * @since  {DEPLOY_VERSION}
+ */
+class RelationObserver extends AbstractDataMapperObserver
+{
+	/**
+	 * Property deleteTempDataset.
+	 *
+	 * @var  DataSet
+	 */
+	protected $deleteTempDataset;
+
+	/**
+	 * Property parentTable.
+	 *
+	 * @var  Table
+	 */
+	protected $parentTable;
+
+	/**
+	 * Creates the associated observer instance and attaches it to the $observableObject
+	 *
+	 * @param   JObservableInterface $observableObject The observable subject object
+	 * @param   array                $params           Params for this observer
+	 *
+	 * @return  JObserverInterface
+	 *
+	 * @since   3.1.2
+	 */
+	public static function createObserver(JObservableInterface $observableObject, $params = array())
+	{
+		$observer = new static($observableObject, $params);
```

Comment:
基本上他目前的設計就是這樣

2015-07-31 12:25 GMT+08:00 Leo <notifications@github.com>:

> In src/DataMapper/Observer/RelationObserver.php
> <https://github.com/smstw/windwalker-joomla-rad/pull/100#discussion_r35945802>
> :
>
> > +	 */
> > +	protected $parentTable;
> > +
> > +	/**
> > +	 * Creates the associated observer instance and attaches it to the $observableObject
> > +	 *
> > +	 * @param   JObservableInterface $observableObject The observable subject object
> > +	 * @param   array                $params           Params for this observer
> > +	 *
> > +	 * @return  JObserverInterface
> > +	 *
> > +	 * @since   3.1.2
> > +	 */
> > +	public static function createObserver(JObservableInterface $observableObject, $params = array())
> > +	{
> > +		$observer = new static($observableObject, $params);
>
> 看起來是 Joomla 原始碼的問題
>
> JTableObserver::__construct 裡面的 $table->attachObserver($this);
> 這個操作表示 $table 應該要 implement JObservableInterface...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/smstw/windwalker-joomla-rad/pull/100/files#r35945802>.
>



-- 
Best Regard

Simon Asika | Kwang Po Tung | 董冠伯
------------------------------------------------
- Founder: Asikart Extension <http://ext.asikart.com/>
- CTO: SMS Taiwan <http://www.bm-sms.com.tw/>

- Mobile: 886-928-565-747
- Github: https://github.com/asika32764